### ISSUE TITLE
Replace 'Virtus' with 'dry-types' for Coercion

### ISSUE THREAD

**Issue by b2e9f523** [G5/storext#37]
Hey,

first of all, thanks for writing this great gem!
Are there any plans to replace the 'Virtus' dependency with the much better maintained [dry-types](https://github.com/dry-rb/dry-types) coercion engine? (Same creator)
It's even the default coercion backend for the Trailblazer architecture.

Btw, the creator of Virtus explains why he started developing the dry-types gem: [Virtus to be abandoned by its creator?](https://www.reddit.com/r/ruby/comments/3sjb24/virtus_to_be_abandoned_by_its_creator/)

Would love to hear your thoughts about it.

**Comment by ramontayag** [comment]
Hello @b2e9f523 - glad you find it useful!

I'm familiar with the Dry gems and use them for other projects. I considered moving but dry-types is not `1.x` yet. I'm concerned with breaking changes.

I don't see any reason other than that. It seems the feature set storext needs can be found in dry-types. Perhaps we can keep this open until dry-rb reaches 1.x?

**Comment by b2e9f523** [comment]
Yes, better wait until dry-types is mature enough to rely on without facing breaking changes.

**Comment by FunkyloverOne** [comment]
The time has come :tada: 
https://dry-rb.org/news/2019/04/23/dry-types-and-dry-struct-1-0-0-released/

Also, they will release `dry-validation` and `dry-schema` 1.0.0 soon, if those might be needed here as well.

**Comment by ramontayag** [comment]
That's great!

If anyone wants to take a look at upgrading, please do - I won't be able to do so for a while. All the conversion happens through a [proxy object](https://github.com/G5/storext/blob/f56deb96b3c5c98d72d59b0b5d0a4e26987bf194/lib/storext/class_methods.rb#L77) that is created per class that uses Storext.

**Comment by swrobel** [comment]
FYI, [virtus 2.0](https://github.com/solnic/virtus/blob/master/Changelog.md) was just released. It seems that [this gem's dependency](https://github.com/G5/storext/blob/master/storext.gemspec#L20) on it is unbounded, so compatibility could possibly be broken by this major version bump.
